lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/openturns/LinearModelValidation.hxx

```diff
@@ -0,0 +1,99 @@
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_LINEARMODELVALIDATION_HXX
+#define OPENTURNS_LINEARMODELVALIDATION_HXX
+
+#include "openturns/Sample.hxx"
+#include "openturns/LinearModelResult.hxx"
+#include "openturns/MetaModelValidation.hxx"
+#include "openturns/SplitterImplementation.hxx"
+#include "openturns/LeaveOneOutSplitter.hxx"
+#include "openturns/KFoldSplitter.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class LinearModelValidation
+ *
+ * The linear model analysis
+ */
+
+class OT_API LinearModelValidation :
+  public MetaModelValidation
+{
+  CLASSNAME
+
+public:
+  /** Default constructor */
+  LinearModelValidation();
+
+  /** Parameter constructor based on result */
+  explicit LinearModelValidation(const LinearModelResult & linearModelResult);
+
+  /** Parameter constructor based on result and leave-one-out */
+  explicit LinearModelValidation(const LinearModelResult & linearModelResult,
+                                 const LeaveOneOutSplitter & splitter);
+
+  /** Parameter constructor based on result and K-Fold */
+  explicit LinearModelValidation(const LinearModelResult & linearModelResult,
+                                 const KFoldSplitter & splitter);
+
+  /** Virtual constructor */
+  LinearModelValidation * clone() const override;
+
+  /** String converter */
+  String __repr__() const override;
+
+  /** Linear model accessor */
+  LinearModelResult getLinearModelResult() const;
+
+  /** Get the splitter */
+  SplitterImplementation getSplitter() const;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const override;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv) override;
+
+private:
+
+  /** Initialize the object*/
+  void initialize();
+
+  /** linear model result */
+  LinearModelResult linearModelResult_;
+
+  /** Splitter parameter */
+  SplitterImplementation splitter_;
+  
+  /** Compute cross-validation leave-one-out predictions */
+  static Sample ComputeMetamodelLeaveOneOutPredictions(const LinearModelResult & linearModelResult, 
+                                                       const LeaveOneOutSplitter & splitter);
+
+  /** Compute cross-validation K-Fold predictions */
+  static Sample ComputeMetamodelKFoldPredictions(const LinearModelResult & linearModelResult, 
```

Comment:
Same remarks as for `chaos`.

However if we think about `static` methods, maybe we can implement them at `MetaModelValidation` level with different arguments ?  For example something like
```cxx
static Sample ComputeMetamodelKFoldPredictions(const Sample & inputSample, const Sample outputSample,
                                               const Sample & residualSample, const FunctionCollection & basis,
                                               const Function & transformation)

```

Reply:
I assume that you want to factor the part of this method which is common to LinearModelResult and FunctionalChaosResult. This would be straightforward if we could convert a FunctionalChaosResult into a LinearModelResult, but we cannot (as you know :)). Perhaps we could create a LeastSquaresResult, which could be used by both classes? After all, a regression is a particular type of least squares and a FCResult too.

Reply:
my idea is to factor yes. IMHO we can do it easily without creating complex objects but directly by defining a static method with the above signature, as both FCE/LM finally relies on these object to compute LOO/KFold cross validations...

Reply:
Ok. Fixed at b4a6ce9668ed342cb94f82709bada6f3119b1f33.